1.18.2 dev

Lunki51 opened the pull request:
Hi really like your project and the original mod. I would love to help you. To get a first hand on the project I made the JEI implementation which is not yet quite finished but contain the basic classes. Feel free to ask for changes in what i've done.

Implemented a basic JEI support
Implemented devices:
-Distillery
-Kettle
-Spinning Wheel
-Witch Cauldron
-Witch Oven

Still missing:
-Rites support
-Kettle and Witch Cauldron craft GUI textures
-Required power info on each craft
-Text description of certain objects like filled chalice

Commit:
JEI Implementation for WitchOven

Commit:
Size change

Commit:
Distillery JEI integration
Remove witch_oven_craft texture because it is useless

Commit:
Spinning Wheel JEI integration
SPINNING_WHEEL RecipeType changed to SpinningWheelRecipe instead of KettleRecipe

Commit:
Witch Cauldron JEI integration
Cauldron GUI texture still missing

Commit:
Kettle JEI integration
Kettle GUI texture still missing

Favouriteless commented:
Appreciated, I'll take a look when I wake up tomorrow.

I'll probably be making a lot of changes to the GUIs but this is a big help, tysm. Please ask on discord before trying to help with anything else though, I don't want to have an awkward situation where somebody wrote a lot of code for something I'd already started working on.

Lunki51 commented:
Hey yeah that's me on discord.
Sry I didn't asked on what to do. I wasn't even sure that I'll actually finish something.
So to start I just picked something from your todo-list. 
I'll keep pushing the thing that are still missing on the branch before you pull

Favouriteless committed:
Copyright notices & move

- Moved categories into their own package.
- Copyright notice on all files.

Favouriteless committed:
Reformatting category classes.

Favouriteless committed:
AbstractCreateItemRite

- AbstractCreateItemRite now takes an ItemStack[] in constructor and has methods for getResultItems and setupItemNbt.
- replaceItem method moved to AbstractRite.
- Refactored all AbstractCreateItem rites.
- Rites of broiling and talisman binding are no longer AbstractCreateItemRites

Favouriteless committed:
AbstractCreateItemRite

- AbstractCreateItemRite now takes an ItemStack[] in constructor and has methods for getResultItems and setupItemNbt.
- replaceItem method moved to AbstractRite.
- Refactored all AbstractCreateItem rites.
- Rites of broiling and talisman binding are no longer AbstractCreateItemRites

Commit:
Merge remote-tracking branch 'origin/1.18.2-dev' into 1.18.2-dev

Favouriteless committed:
setupItemNBT checks

Commit:
Merge remote-tracking branch 'origin/1.18.2-dev' into 1.18.2-dev

Commit:
Rite JEI integration
Added localization of strings

Commit:
Renaming some vars

Commit:
Now using patchouli textures for the magic circles

Commit:
Power Cost is displayed for everything
Using AbstractRite instead of AbstractCreateItemRite
Dummy texture for Kettle and Cauldron

Commit:
Edited WitchOvenCategory to include the byproduct and the craft result
Edited WitchOvenRecipe to take an array as ingredients instead of a single object. Because minecraft:logs includes things like crimson_stem that are not part of the recipe

Favouriteless committed:
Revert "Edited WitchOvenCategory to include the byproduct and the craft result"

This reverts commit 92004d54091ebcf18a31cc5a012281a50ee8e156.
